wordvecspace/annoy.py

```diff
@@ -85,23 +129,234 @@ def get_distances(self, row_words_or_indices: Union[int, str, np.ndarray],\
+        array([15445, 14669, 12198,  3801, 22196], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, combination=True, combination_method='set_union',
+        ... examine_k=500, weights=[0.2, 0.5], include_distances=True)
+        (array([15445, 14669, 12198,  3801, 22196], dtype=uint32), array([0.4889, 0.4891, 0.4924, 0.4938, 0.4944], dtype=float32))
+        >>> u = wv.get_nearest(vecs, k=1000, combination=True, combination_method='set_union',
+        ... examine_k=1500, weights=[0.2, 0.5])
+        >>> i = wv.get_nearest(vecs, k=1000, combination=True, combination_method='set_intersect',
+        ... examine_k=1500, weights=[0.2, 0.5])
+        >>> u == i
+        False
+        """
+
+        if not combination:
+            return self._get_nearest(v_w_i, k, include_distances)
+        elif combination_method == 'set_intersect':
+            return self._get_set_intersect(v_w_i, k, examine_k, metric, weights, include_distances)
+        elif combination_method == 'set_union':
+            return self._get_set_union(v_w_i, k, examine_k, metric, weights, include_distances)
+        elif combination_method == 'vector':
+            return self._get_resultant_vector_nearest(v_w_i, k, weights, metric, include_distances)
+        else:
+            raise InvalidCombination()
+
+    def _get_resultant_vector_nearest(self, v_w_i, k, weights, metric, include_distances):
+        """Retrieves nearest indices based on the resultant vector.
+
+        Algorithm:
+        i) Compute the resultant vector by summing per dimension across vectors
+        ii) Commpute distances based on this resultant vector
+        iii) Return nearest indices after sorting
+
+        Tests:
+        >>> wv = WordVecSpaceAnnoy(DATAFILE_ENV_VAR)
+        >>> vecs = wv.get_vectors(['laptop', 'phone'])
+        >>> wv.get_nearest(vecs, k=5, combination=True, combination_method='vector',
+        ... include_distances=True)
+        (array([10101, 27639, 25905, 28303,  8971], dtype=uint32), array([0.0529, 0.0622, 0.1041, 0.1386, 0.1539], dtype=float32))
+        >>> wv.get_nearest(vecs, k=5, combination=True, combination_method='vector',
+        ... include_distances=True)
+        (array([10101, 27639, 25905, 28303,  8971], dtype=uint32), array([0.0529, 0.0622, 0.1041, 0.1386, 0.1539], dtype=float32))
+        """
+
+        v = self._check_vec(v_w_i)
+
+        res = None
+        if not weights:
+            weights = np.ones(len(v_w_i))
+        else:
+            weights = np.array(weights)
 
-        index = self._check_index_or_word(v_w_i)
+        if isinstance(v, (list, np.ndarray)):
+            resultant_vec = (v * weights[:, None]).sum(axis=0, dtype=np.float32)
+            res, distances = self.ann.get_nns_by_vector(resultant_vec, k, include_distances=True)
 
-        return self.ann.get_nns_by_item(index, k)
+        if include_distances:
+            return (np.array(res, dtype=np.uint32), np.array(distances, dtype=np.float32))
+        else:
```

Comment:
We do not need an else here anyway if it passes if condition it return from there itself. we can reduce one indent for the next return